docs/userGuide/deployingTheSite.md

```diff
@@ -329,10 +329,172 @@ Here are the steps to set up Netlify:
+
+    <include src="screenshot.md" boilerplate var-alt="Get Surge token" var-file="surgeToken.png" inline />
+
+1. In the repo of your markbind site, create a new secret by going to "Settings"->"Secrets" and naming it as `SURGE_TOKEN` and setting its value to the value of the generated surge token.
+
+    <include src="screenshot.md" boilerplate var-alt="Add Surge token" var-file="surgeAddToken.png" inline />
+
+1. Commit and push the following 2 files into your markbind site repo, in the directory `<PROJECT_ROOT>/.github/workflows/`.
+
+<panel header="`receivePR.yml` File" type="seamless">
+
+{% raw %}
+```yml {.no-line-numbers}
+name: Receive Markbind PR
+
+# read-only
+# no access to secrets
+on:
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:        
+      - uses: actions/checkout@v2
+      - name: Install Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 10
+      - name: Build Markbind website
+        if: ${{ success() && github.event_name == 'pull_request' }}
+        run: |
+          npm install -g markbind-cli
+          markbind build
+      - name: Save PR number and HEAD commit
+        if: ${{ success() && github.event_name == 'pull_request' }}
+        run: |
+          mkdir -p ./pr
+          echo ${{ github.event.number }} > ./pr/NUMBER
+          echo ${{ github.event.pull_request.head.sha }} > ./pr/SHA
+      - name: Upload artifacts 
+        if: ${{ success() && github.event_name == 'pull_request' }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: markbind-deployment
+          path: |
+            ./_site
+            ./pr
+```
+{% endraw %}
+
+</panel>
+
+<panel header="`previewPR.yml` File" type="seamless">
+
+{% raw %}
+```yml {.no-line-numbers}
+name: Deploy PR Preview
+
+# Runs after PR is received and build by markbind-cli
+# Has access to repo secrets
+on:
+  workflow_run:
+    workflows: ["Receive Markbind PR"]
+    types:
+      - completed
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Deploying to surge
+    steps:
+      - uses: actions/checkout@v2
+      - name: Download artifact
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          workflow: receivePR.yml
+          run_id: ${{ github.event.workflow_run.id }}
+          name: markbind-deployment
+          path: .
+      - name: Display structure of downloaded files
+        run: ls -R
+      - name: Extract PR number
+        id: pr-number
+        run: echo '::set-output name=ACTIONS_PR_NUMBER::'$(cat ./pr/NUMBER)
+      - name: Install Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 10
+      - name: Inject slug/short variables
+        uses: rlespinasse/github-slug-action@v3.x
+      - name: Build PR preview url
+        id: pr-url
+        run: echo '::set-output name=ACTIONS_PREVIEW_URL::'https://pr-${{ steps.pr-number.outputs.ACTIONS_PR_NUMBER }}-${{ env.GITHUB_REPOSITORY_SLUG_URL }}.surge.sh/
```

Comment:
Hmm I think that if we remove `rlespinasse/github-slug-action@v3.x`, it will add some more complexity since there needs to be some regex script to extract the repoSlug. I am not sure if there is a way to simplify the surge url while ensuring that it is still as general as possible.

I can add a note to say that the user can make modifications to the surge URL as needed to a simpler one, together with a short code snippet showing the relevant changes. What do you think @ang-zeyu ?

Reply:
not too familiar with surge's specifics - is `${{ env.GITHUB_REPOSITORY_SLUG_URL }}` necessary / convention for pr previewing in surge?

my view is that this could be easily configured to something unique if the user wants to (not limited to `owner + repo` format).
i.e. just `https://pr-${{ steps.pr-number.outputs.ACTIONS_PR_NUMBER }}-xxxx-anything-xxxx.surge.sh`.
Would that work?

Reply:
One way is that the user can have a `SURGE_URL` secret, so the PR URL would be something like `https://pr-${{ steps.pr-number.outputs.ACTIONS_PR_NUMBER }}-${{ secrets.SURGE_URL }}.surge.sh`. 

This would eliminate the need for `rlespinasse/github-slug-action@v3.x` in the code. However, this would be slightly more inconvenient for the user.

What do you think about this approach?

Reply:
Hmm was thinking of just highlighting that the specific line in the `.yml` should be edited per the project as needed before committing. (comment / warning box / highlight)
Since folks who are using pull requests, github actions, github likely have the know how =P

Reply:
Okay I have added in the userGuide a new variable `PR_URL` which the user can modify. I have also added a note to highlight to the user that they have to edit the `PR_URL` according to their needs.